app/components/search-facet-worktype/component.js

```diff
@@ -1,26 +1,51 @@
 import Ember from 'ember';
-import { termsFilter, invertTermsFilter } from 'ember-share/utils/elastic-query';
+import { termsFilter, invertTermsFilter, getUniqueList } from 'ember-share/utils/elastic-query';
 
 export default Ember.Component.extend({
+
+    init() {
+        this._super(...arguments);
+        this.send('toggle', this.get('options.param'));
+    },
+
     selected: Ember.computed('filter', function() {
         return invertTermsFilter(this.get('key'), this.get('filter'));
     }),
 
-    buildQueryObject(selected) {
-        let key = this.get('options.queryKey') || this.get('key');
-        return {key: key, selected: selected, param2: true, filterType: termsFilter};
+    buildQueryObjectCombine(selected) {
+        let key = this.get('key');
+        let currentFilter = this.get('options.facetFilters')[key] ? invertTermsFilter(key, this.get('options.facetFilters')[key])[0] : [];
+        let value = !selected[0] ? [] : selected;
+        let newValue = getUniqueList(Array.prototype.concat(value, currentFilter));
+        let newFilter = termsFilter(key, newValue, this.get('options.raw'));
+        return [newFilter, newValue];
+    },
+
+    buildQueryObjectMatch(selected) {
+        let key = this.get('key');
+        let newValue = !selected[0] ? [] : selected;
+        let newFilter = termsFilter(key, getUniqueList(newValue), this.get('options.raw'));
+        return [newFilter, newValue];
     },
 
     actions: {
-        toggle(type) {
+        toggle(type, match=true) {
```

Comment:
Same thing: any filter that can be added by clicking on a search result will have to use the combine method.